sympy/external/ntheory.py

```diff
@@ -153,3 +153,49 @@ def kronecker(x, y):
     if s % 2 and x % 8 in [3, 5]:
         sign = -sign
     return sign * jacobi(x, y)
+
+
+def iroot(y, n):
+    if y < 0:
+        raise ValueError("y must be nonnegative")
+    if n < 1:
+        raise ValueError("n must be positive")
+    if y in (0, 1):
+        return y, True
+    if n == 1:
+        return y, True
+    if n == 2:
+        x, rem = mlib.sqrtrem(y)
+        return int(x), not rem
+    if n >= y.bit_length():
+        return 1, False
+    # Get initial estimate for Newton's method. Care must be taken to
+    # avoid overflow
+    try:
+        guess = int(y**(1./n) + 0.5)
+    except OverflowError:
+        exp = math.log(y, 2)/n
```

Comment:
I thought that this call to `log` might overflow for large values of `y` but that does not seem to be the case for Python (`>= 3.8`).

I'm not sure if that is something that could be depended on for any implementation of Python though. I just tested with pypy though and it seems to work.

Looking at the docs I see that there is a log2 function since Python 3.3:
```
Return the base-2 logarithm of x. This is usually more accurate than log(x, 2).
```

Reply:
Maybe that would more reliably work for large integers...

Reply:
Changed from `log` to `log2`